### ISSUE TITLE
Can't seem to get optipng working....

### ISSUE THREAD

**Issue by DumbJoe** [greenshot/greenshot#147]
On the [FAQ](https://getgreenshot.org/faq/is-there-any-way-to-optmize-the-png-files/) it says to edit this line `OptimizePNGCommand=` to where ever you saved the file in, in my case I saved it in the directory as the installed version so it should read: `OptimizePNGCommand=C:\Program Files\Greenshot\optipng.exe` and the commands to it are these: `OptimizePNGCommandArguments="{0}"` which I've edited to `OptimizePNGCommandArguments=-o7` but doesn't work... I ran the program CLi seperate and it works compressing the pngs, but with your app, doesn't seem to run it....

The two lines as shown in the ini file:
```
; Optional command to execute on a temporary PNG file, the command should overwrite the file and Greenshot will read it back. Note: this command is also executed when uploading PNG's!
OptimizePNGCommand=C:\Program Files\Greenshot\optipng.exe
; Arguments for the optional command to execute on a PNG, {0} is replaced by the temp-filename from Greenshot. Note: Temp-file is deleted afterwards by Greenshot.
OptimizePNGCommandArguments=-o7
```

**Comment by Lakritzator** [comment]
First: These lines need to be under the correct section in the init, so I hope they are still there.
The arguments are not correct, as you removed the information where the file is located.

You probably want this:
```
; Optional command to execute on a temporary PNG file, the command should overwrite the file and Greenshot will read it back. Note: this command is also executed when uploading PNG's!
OptimizePNGCommand=C:\Program Files\Greenshot\optipng.exe
; Arguments for the optional command to execute on a PNG, {0} is replaced by the temp-filename from Greenshot. Note: Temp-file is deleted afterwards by Greenshot.
OptimizePNGCommandArguments=-o7 "{0}"
```

Next time please report in our JIRA, as is specified in the template, you are ignoring our terms for creating issues.

**Comment by DumbJoe** [comment]
Oh I see, I was suppose to keep the `"{0}"` and just add it before with a space for the command. Woops...

Both relative and absolute paths work, though for the installed version no idea the relative is from where the ini file is located or from where the program is installed on as I've only tested on the portable version where both are located in the same folder.

Ok so it works, or at least the lower and mid levels do. The higher levels don't seem to work or if they do, seems to be taking a lot longer than it would if you had just manually gone through optipng and throw in the command through commandline interface and worked from there. I've patiently waited for half an hour and it's still processing by the looks of it.... I can go as high as level 5, but level 6 and higher seem to not work as intended...

I checked the location of the temp file and it either is saved there without optimisation or the optimisation is already saved but for some reason hasn't transferred to the directed save location it was specified when screenshotting.

> Next time please report in our JIRA, as is specified in the template, you are ignoring our terms for creating issues.

I wasn't aware there was a separate issue/quesetion tracker for users, just used to reporting it on the place it was downloaded from - in this case github.
